File: DesignPattenStudy/DecoratorPattern/Java/Concrete/EspressoDecorator.java
package DecoratorPattern.Java.Concrete;

import DecoratorPattern.Java.Abstract.Component;
import DecoratorPattern.Java.Abstract.Decorator;

public class EspressoDecorator extends Decorator {

    public EspressoDecorator(Component component) {
        super(component);
    }

    @Override
    public Integer addPrice() {
        return super.addPrice() + 500;
    }

    @Override
    public String showItem() {
        return super.showItem() + " + SteamWater : Espresso";
    }
}

File: DesignPattenStudy/DecoratorPattern/Kotlin/Concrete/EspressoDecorator.kt
package DecoratorPattern.Kotlin.Concrete

import DecoratorPattern.Kotlin.Astract.Component
import DecoratorPattern.Kotlin.Astract.Decorator

class EspressoDecorator(override val component: Component) : Decorator(component) {

    override fun addPrice(): Int = super.addPrice() + 500

    override fun showItem(): String = "${super.showItem()} + SteamWater : Espresso"
}